lib/rules/rule-selector-property-disallowed-list/index.mjs

```diff
@@ -49,14 +49,18 @@ const rule = (primary) => {
 				return;
 			}
 
-			ruleNode.walkDecls((decl) => {
-				const { prop } = decl;
+			ruleNode.walk((node) => {
+				if (node.type !== 'decl') return;
+
+				if (!declarationIsAChildOfRule(node, ruleNode)) return;
```

Comment:
@ybiquitous I think that this check matches the original intent of this rule and fixes the issues for nesting.

The original bug report happend when validation `.foo {}` for `.foo { & input {} }`. This change adds an early return so that nested declarations are only tested against their most immediate ancestor rule. (if that makes sense?)

Reply:
Yes, that makes sense! 👍🏼